### PR TITLE
COOPR-574: CLI support for auth tokens

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
Added CLI support for auth tokens. Changed hierarchy of commands: added AbstractAuthCommand that updates access token in case of this token becomes invalid. For now hierarchy looks like: AbstractAuthCommand implements Command; <Any specific command> extends Abstract Auth Command.

File: coopr-cli/src/main/java/co/cask/coopr/shell/CLIConfig.java [hunk 1]
```diff
@@ -115,9 +122,17 @@ public void setConnection(String host, int port, boolean ssl, String userId, Str
     } else {
       this.port = port;
     }
+    this.ssl = ssl;
     this.uri = URI.create(String.format("%s://%s:%d", ssl ? "https" : "http", host, port));
     RestClientManager.Builder builder = RestClientManager.builder(host, port);
-    builder.ssl(DEFAULT_SSL);
+    builder.ssl(ssl);
+    accessToken = getAccessToken(host, port, ssl);
+    builder.accessToken(new Supplier<AccessToken>() {
```
Comment (jwang47) [inline coopr-cli/src/main/java/co/cask/coopr/shell/CLIConfig.java]: Can use `Suppliers.ofInstance(accessToken)`

File: coopr-cli/src/main/java/co/cask/coopr/shell/command/AbstractAuthCommand.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.io.PrintStream;
+
+/**
+ * Abstract command for updating {@link co.cask.cdap.security.authentication.client.AccessToken}.
+ */
+public abstract class AbstractAuthCommand implements Command {
+
+  private final CLIConfig cliConfig;
+
+  AbstractAuthCommand(CLIConfig cliConfig) {
```
Comment (jwang47) [inline coopr-cli/src/main/java/co/cask/coopr/shell/command/AbstractAuthCommand.java]: protected

File: coopr-rest-client/src/main/java/co/cask/coopr/client/rest/RestClient.java [hunk 3]
```diff
@@ -217,12 +221,18 @@ protected URI buildFullURL(String postfix) {
   }
 
   private Header[] getAuthHeaders() {
```
Comment (jwang47) [inline coopr-rest-client/src/main/java/co/cask/coopr/client/rest/RestClient.java]: Why is this an array rather than a set?

**Comment by rpedanov** [comment]
@jwang47 done

**Comment by jwang47** [comment]
LGTM

**Pull request merged**